hydrolib/core/dflowfm/tim/models.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+from typing import Callable, Dict, List
+
+from pydantic.class_validators import validator
+
+from hydrolib.core.basemodel import ModelSaveSettings, ParsableFileModel
+
+from .parser import TimParser
+from .serializer import TimSerializer, TimSerializerConfig
+
+
+class TimModel(ParsableFileModel):
+    """Class representing a tim (*.tim) file.
+
+    Attributes:
+        data: Dictionary with keys \"comment\" & time as float and value as List of floats".\n
+        serializer_config: TimSerializerConfig
+
+    Raises:
+        ValueError: If the time and values are not parsable to float, or when the amount of columns differs per timeseries.
+    """
```

Comment:
```suggestion
    """Class representing a tim (*.tim) file."""
```

The `Raises` does not need to be documented on class level, only on function level.
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html.

As a team we have decided to document the attributes of a class not on class level, but on attribute level.

So for example:

```python
    serializer_config = TimSerializerConfig()
    """TimSerializerConfig: The serialization configuration for the tim file."""

    comments: List[str]
    """List[str]: A list with the header comment of the tim file."""
    
    timeseries: Dict[float, List[float]]
    """Dict[float, List[float]]: A dictionary containing the time series. The keys are the times and the values are the values corresponding to that time."""

```